operator/operator.go

```diff
@@ -160,6 +162,78 @@ func (operator *Operator) handleBlocks() error {
 	return nil
 }
 
+func (operator *Operator) handleChallenges() error {
+	challenges := make(chan *bindings.ErdstallChallenged)
+	sub, err := operator.contract.WatchChallenged(nil, challenges, nil, nil)
+	if err != nil {
+		return fmt.Errorf("creating challenge subcription: %w", err)
+	}
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case err := <-sub.Err():
+			return fmt.Errorf("subscription error: %w", err)
+
+		case _c := <-challenges:
+			c := challengedEvent(*_c)
+			log.Warnf("Operator.handleChallenges: Incoming challenge %v", c)
+
+			if err := operator.handleChallengedEvent(c); err != nil {
+				log.Errorf("Operator.handleChallenges: Failed to handle challenged event %v: %v", c, err)
+			}
+		}
+	}
+}
+
+func (operator *Operator) handleChallengedEvent(c challengedEvent) error {
+	ctx, cancel := createDefaultContext()
+	defer cancel()
+
+	tr, err := operator.ethClient.NewTransactor(ctx, big.NewInt(0), eth.DefaultGasLimit, operator.ethClient.Account())
+	tr.Context = ctx
```

Comment:
move below `if err` check or this might panic due to a `nil` `tr`.

Reply:
done